mbed_greentea/mbed_report_api.py

```diff
@@ -768,6 +770,10 @@ def exporter_memory_metrics_csv(test_result_ext, test_suite_properties=None):
         for test_suite_name in test_results:
             test = test_results[test_suite_name]
 
+            # If test didn't pass, memory stats doesn't make much sense
+            if test['single_test_result'] != TEST_RESULT_OK:
+                continue
```

Comment:
@studavekar Seems like this is pretty harmless to leave in anyway right? Maybe the test failed due to a stack overflow...